### DISCUSSION TITLE
getStaticProps with amp configuration runs only while build stage

### DISCUSSION THREAD

**Discussion by callsmasterofficial** [vercel/community#1173] [Help] [answered]
### Summary

when amp config is set to 'hybrid', getStaticProps in dynamic url runs only at build stage. How can we handle fallback=true in getStaticPath with getStaticProps for amp page?

**Comment by amyegan** [comment, marked as answer]
Hi @callsmasterofficial! It is normal for `getStaticProps` to be used only at build time. But I think there's something else going on here that I'm missing. Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can get a better view of the problem you're seeing?

**Reply by callsmasterofficial** [reply]
Hi @amyegan ,
Here is sample code what we are using:- 

![3rd](https://user-images.githubusercontent.com/78918211/209334832-3e7afe6c-9a6f-4c24-baaf-2fc85b6178eb.png)


and the output of this is as without amp 
![3rd](https://user-images.githubusercontent.com/78918211/209334700-1cd9cea7-ddc2-473f-9d26-452f92ff6a57.png)


then when we tried with amp version of the page 
![3rd](https://user-images.githubusercontent.com/78918211/209334543-74534b62-7b8d-407a-b801-b5bdd0be9f30.png)


Nothing is being returned from getStaticProps

**Reply by amyegan** [reply]
When you use [`fallback: true`](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-true), you need to return the paths you want to pre-render at build time from `getStaticPaths`.

> The paths returned from `getStaticPaths` will be rendered to HTML at build time by `getStaticProps`.
>
>The paths that have not been generated at build time will not result in a 404 page. Instead, Next.js will serve a [“fallback”](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-pages) version of the page on the first request to such a path. Web crawlers, such as Google, won't be served a fallback and instead the path will behave as in [fallback: 'blocking'](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-blocking).

On the [fallback](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-pages) version of a page, the page’s props will be empty. The [router](https://nextjs.org/docs/api-reference/next/router) can be used to detect if the fallback is being rendered (`router.isFallback` will be true) so you can control what else will be displayed.

If you change your `getStaticPaths` to 
```
export async function getStaticPaths() {
  return {
    paths: [
      {
        params: {
          page: "contact",
        },
      }
    ],
    fallback: true,
  };
}
```
then it will display "You are at contact page" for both localhost:3000/contact and localhost:3000/contact?amp=1, but will still use the fallback for all other amp routes.

The Next.js site has a good example of populating dynamic routes so you don't need to manually configure `getStaticPaths` output to include every possible page on your site: https://nextjs.org/learn/basics/dynamic-routes/page-path-external-data

**Reply by callsmasterofficial** [reply]
Hi 
Thanks for such an elaborated answer.
`getStaticPaths` will  run only once at build time. At the build time `getStaticPaths` will fetch all the page slug available and pre-render those page.
 Suppose we have our site running live after building the project and then after we need to add new page from our project dashboard. In that case how can we publish the new page?

**Reply by amyegan** [reply]
To create or update individual static pages after you’ve built your site, you can use [Incremental Static Regeneration](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration)
